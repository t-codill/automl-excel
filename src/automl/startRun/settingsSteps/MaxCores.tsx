import * as React from "react";
import { FormTextInput } from "../../components/Form/FormTextInput";
import { Validators } from "../../components/Form/Validators";
import { LabelWithCallout } from "../../components/Label/LabelWithCallout";
import { ISettingsStepParams } from "./ISettingsStepParams";

const maxCores: React.FunctionComponent = () => {
    return <div className="ms-Grid" dir="ltr">
        <div className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6">
            <LabelWithCallout required={false} htmlFor={"maxCoresInput"} labelText="Max cores per iteration">
                Maximum number of threads to use for a given training iteration.
            </LabelWithCallout>
        </div>
        <div className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6">
            <FormTextInput<ISettingsStepParams, "maxCores">
                field="maxCores"
                placeholder="Max cores per iteration"
                ariaLabel="Please enter the max number of cores here"
                type="number"
                validators={[
                    Validators.isInteger("Max cores per iteration should be an integer"),
                ]}
            />
        </div>
    </div>;
};

export { maxCores as MaxCores };
