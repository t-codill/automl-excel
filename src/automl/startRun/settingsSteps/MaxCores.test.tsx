import { shallow } from "enzyme";
import * as React from "react";
import { MaxCores } from "./MaxCores";

describe("Max Cores", () => {
    it("should render ", () => {
        const tree = shallow(
            <MaxCores />
        );
        expect(tree)
            .toMatchSnapshot();
    });
});
