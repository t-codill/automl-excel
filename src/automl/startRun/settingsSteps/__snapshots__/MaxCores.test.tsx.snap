// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Max Cores should render  1`] = `
<div
  className="ms-Grid"
  dir="ltr"
>
  <div
    className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6"
  >
    <LabelWithCallout
      htmlFor="maxCoresInput"
      labelText="Max cores per iteration"
      required={false}
    >
      Maximum number of threads to use for a given training iteration.
    </LabelWithCallout>
  </div>
  <div
    className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6"
  >
    <FormTextInput
      ariaLabel="Please enter the max number of cores here"
      field="maxCores"
      placeholder="Max cores per iteration"
      type="number"
      validators={
        Array [
          [Function],
        ]
      }
    />
  </div>
</div>
`;
